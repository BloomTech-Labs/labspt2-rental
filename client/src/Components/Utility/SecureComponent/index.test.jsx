import React from 'react';
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import SecuredComponent from './index';


const fakeUser = {
    username: 'Grape Ape',
    permissions: {
        read: true,
        write: false,
    }
}

describe('<SecuredComponent />', () => {
    it('should render with appropriate props', () => {
        const securedComponent = mount(
            <SecuredComponent permission="read" user={fakeUser}>
                <div>It must have children</div>
            </SecuredComponent>
        );
        expect(securedComponent.exists()).toBeTruthy()
        securedComponent.unmount()
    })
    it('should render children if permission matches', () => {
        const securedComponent = mount(
            <SecuredComponent permission="read" user={fakeUser}>
                <div>child1</div>
                <div>child2</div>
                <div>child3</div>
            </SecuredComponent>
        );
        expect(securedComponent.find('div').length).toBe(3)
        securedComponent.unmount()
    });

    it('should have null if permission is incorrect', () => {
        const securedComponent = mount(
            <SecuredComponent permission="write" user={fakeUser}>
                <div>child1</div>
                <div>child2</div>
                <div>child3</div>
            </SecuredComponent>
        );
        expect(securedComponent.find('div').length).toBe(0)
        securedComponent.unmount()
    })

    it('snapshot', () => {
        const securedComponentSnapshot = renderer.create(
            <SecuredComponent permission="read" user={fakeUser}>
                <div>It must have children</div>
            </SecuredComponent>
        );
        expect(securedComponentSnapshot.toJSON()).toMatchSnapshot()
    })
})