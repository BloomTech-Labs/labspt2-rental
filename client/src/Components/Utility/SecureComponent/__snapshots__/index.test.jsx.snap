// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecuredComponent /> snapshot 1`] = `
<div>
  It must have children
</div>
`;
